services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2ImplTest.java

```diff
@@ -454,6 +455,67 @@ public void testStartOrResumeMultipartRequestWithUnsupportedOperationException()
 		verify(mockMultipartUploadDAO, never()).createUploadStatus(mockCreateMultipartRequest);
 	}
 
+	@Test
+	public void testGetCachedUploadOrRestartNewUpload() {
+		boolean forceRestart = false;
+		when(mockMultipartUploadDAO.getUploadStatus(any(), any())).thenReturn(null);
+
+		// Call under test
+		Optional<CompositeMultipartUploadStatus> status = manager.getCachedUploadOrRestart(user, mockRequest, forceRestart);
+
+		assertTrue(status.isEmpty());
+	}
+
+	@Test
+	public void testGetCachedUploadOrRestartComplete() {
+		boolean forceRestart = false;
+		when(mockMultipartUploadDAO.getUploadStatus(any(), any())).thenReturn(mockCompositeStatus);
+
+		when(mockCompositeStatus.getMultipartUploadStatus()).thenReturn(mockStatus);
+		when(mockStatus.getState()).thenReturn(MultipartUploadState.COMPLETED);
+		when(mockCloudDaoProvider.getCloudServiceMultipartUploadDao(any())).thenReturn(mockCloudDao);
+		when(mockCloudDao.doesObjectExist(any(), any())).thenReturn(true);
+
+		// Call under test
+		Optional<CompositeMultipartUploadStatus> status = manager.getCachedUploadOrRestart(user, mockRequest, forceRestart);
+
+		assertEquals(mockCompositeStatus, status.get());

```

Comment:
you probably want to add some verify to make suer you are using the right parameters in your calls